### PR TITLE
sync scripts for OIDC identities #272

### PR DESCRIPTION
**Pull request by dciangot**
https://github.com/dmwm/CMSRucio/issues/272

### CODE REVIEW

**Comment by dciangot** [comment]
Please notice that I have also introduced a small change in the x509 base sync script, to avoid it to delete OIDC identities when checking

**Comment by dciangot** [comment]
Included the oidc sync script into the x509 .sh as suggested in https://github.com/dmwm/CMSRucio/issues/261

File: helm/rucio-cron-jobs/templates/cronjobs.yaml [hunk 1]
```diff
@@ -83,5 +83,29 @@ spec:
                     secretKeyRef:
                       name: "rucio-secrets"
                       key: monit_token
+                - name: IAM_SERVER
```
Comment (ericvaandering) [inline helm/rucio-cron-jobs/templates/cronjobs.yaml:86]: I don't think the server and client_ID need to be secret, right? We should just be able to make them parameters set directly in the Rucio-flux repository.

File: docker/CMSRucioClient/scripts/k8s_sync_users_links.sh [hunk 2]
```diff
@@ -33,3 +33,20 @@ fi
 
 echo "Creating links"
 ./cmslinks.py --overwrite --disable
+
+# Preparing oidc-agent to get/refresh token to retrieve OIDC user information
+export OIDC_CONFIG_DIR=$HOME/.oidc-agent
```
Comment (ericvaandering) [inline docker/CMSRucioClient/scripts/k8s_sync_users_links.sh:38]: How does .oidc-agent get created/filled?

File: docker/CMSRucioClient/examples/rucio-int-oidc.cfg [hunk 3]
```diff
@@ -0,0 +1,13 @@
+[common]
```
Comment (ericvaandering) [inline docker/CMSRucioClient/examples/rucio-int-oidc.cfg:1]: This file is not used, right? Just an example for someone who does want to authenticate by OIDC? 

I'd suggest changing this to the production server or adding rucio-oidc.cfg which uses the production server.

**Review by ericvaandering — CHANGES_REQUESTED** [review]

File: docker/CMSRucioClient/examples/rucio-int-oidc.cfg (hunk 3 above)
Comment (dciangot) [reply docker/CMSRucioClient/examples/rucio-int-oidc.cfg:1]: yes, it's not used. I'm adding rucio-oidc.cfg in there.

File: docker/CMSRucioClient/scripts/k8s_sync_users_links.sh (hunk 2 above)
Comment (dciangot) [reply docker/CMSRucioClient/scripts/k8s_sync_users_links.sh:38]: if it does not exists, the oidc-agent will create it.

Comment (dciangot) [reply docker/CMSRucioClient/scripts/k8s_sync_users_links.sh:38]: basically to store configuration for the token renewal

**Review by ericvaandering — APPROVED** [review]

**Pull request merged**